spec/schema.json

```diff
@@ -25,7 +25,8 @@
             "in": "path",
             "required": true,
             "schema": {
-              "type": "string"
+              "type": "string",
+              "pattern": "^[A-Za-z0-9]{8}$"
```

Comment:
I think it's only the first two characters that can be letters, the last six must be numbers

Reply:
This matches the AC in the linked story. I think @markpit said that some (obscure) company numbers have three initial characters so we just went for the generic 8 character alpha-numeric string. We can ask an analyst though.

Reply:
I'm reasonably confident that a company with three initial letters won't get far in most of webfiling.
I had a quick scan of what regexes are currently in use. Something like '[A-Z0-9]{2}[0-9]{6}' is common though some places go as far as listing the two-letter combinations that are valid.

Reply:
i got confused, you can have companies with 1 or 2 letters by the looks of it, but analyst should be able to confirm. This is what we had in extensions - https://github.com/companieshouse/extensions-web/blob/master/src/controllers/company.number.controller.ts#L40

Reply:
No doubt CHIPS handles more company types than can file online

Reply:
Mmm. Flakey regex. That would allow 2 letters plus 8 digits which is definitely invalid.

Reply:
Happy with whatever the analyst confirms. Just need to remember to update the JIRA story if the implementation diverges from it.

Reply:
First 2 set to be either letters or numbers, then numbers after that; updated unit tests.

Reply:
Confirmed by analyst? If yes, please update the JIRA AC 4.

Reply:
Now confirmed by analyst (and architect). @mattch1 please can you amend the implementation to match.

Reply:
Upper and lower case will be allowed. Code changed to reflect this so resolving.